src/pip/_internal/build_env/_base.py

```diff
@@ -0,0 +1,134 @@
+import logging
+import os
+import pathlib
+from types import TracebackType
+from typing import TYPE_CHECKING, Iterable, List, Optional, Set, Tuple, Type
+
+from pip._vendor.packaging.requirements import Requirement
+from pip._vendor.packaging.version import Version
+from pip._vendor.typing_extensions import Protocol
```

Comment:
Gosh I never realised we’re already vendoring typing_extensions.

Reply:
It was a fun one to realise that it's easy to add. :)